src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      Object.assign(state, actions[i].extraData);
+    } else if (actions[i].type === 'removeProperties') {
+      for (let j = 0; j < actions[i].keysToRemove.length; j++) {
+        for (let k = 0; k < Object.keys(state).length; k++) {
+          if (actions[i].keysToRemove[j] === Object.keys(state)[k]) {
+            delete state[Object.keys(state)[k]];
+          }
+        }
+      }
+    } else {
```

Comment:
please try to avoid large nesting of loops, the more nesting, the more complex the code.
You should create inside "else if" only one loop and that will be enough.